modules/persistence/tests/services/connector.test.ts

```diff
@@ -0,0 +1,140 @@
+import { ObjectID } from 'bson';
+import { db, insert, upsert } from '../../src/services/connector';
+
+const mockConnect = jest.fn();
+const mockDb = jest.fn();
+const mockCollection = jest.fn();
+const mockInsertMany = jest.fn();
+const mockUpdateOne = jest.fn();
+const mockClose = jest.fn();
+
+// below is a "jest mock" of a mongodb client
+// TODO: update this test module to work with MongoDB memory server
+//       - create test db and verify connector's calls update memory server
+
+jest.mock('mongodb', () => ({
+  MongoClient: class MongoClient {
+    constructor() {
+      console.log('constructor');
+    }
+    connect() {
+      return mockConnect();
+    }
+    async db(...args) {
+      mockDb(...args);
+      return this;
+    }
+    collection(collection) {
+      mockCollection(collection);
+      return this;
+    }
+    async insertMany() {
+      return mockInsertMany();
+    }
+    async updateOne(...args) {
+      return mockUpdateOne(...args);
+    }
+    close() {
+      mockClose();
+    }
+  },
+}));
+
+describe('Connector module', () => {
+  // mock process env. store original env and restore
+  const env = process.env;
+
+  beforeEach(() => {
+    process.env = { ...env };
+  });
+
+  afterEach(() => {
+    process.env = env;
+    jest.restoreAllMocks();
+  });
+
+  describe('db', () => {
+    test('calls client to connect with env variables and returns a mongoDB Db instance', async () => {
+      try {
+        const testEnv = {
+          DB_NAME: 'test-db-name',
+          DB_USER: 'user',
+          DB_PASSWORD: 'password',
+          DB_HOST: 'host',
+        };
+        process.env = {
+          ...env,
+          ...testEnv,
+        };
+        const database = await db();
+        expect(mockConnect).toHaveBeenCalled();
+        expect(mockDb).toHaveBeenCalled();
+        expect(mockDb).toHaveBeenCalledWith(testEnv.DB_NAME);
+      } catch (e) {
+        console.error(e);
+        throw e;
+      }
+    });
+
+    test('returns an error if client fails', async () => {
+      /**
+       * Question for error catching:
+       * why are they not catching in functions (ie. db, insert) when throwing mocked rejections
+       * see connection/db line 32
+       *
+       */
+      mockConnect.mockRejectedValueOnce(new Error('test error') as never);
```

Comment:
@casthewiz i was not seeing this rejected value being logged within [connector ](https://github.com/mongodb/docs-worker-pool/blob/master/modules/persistence/src/services/connector/index.ts#L33)

upsert and insert are written differently (throw in catch vs non) but the `return Promise` seems to bypass the catch block? 

```
try {
  return await insertSession
      .collection(collection)
      .insertMany(docs.map((d) => ({ ...d, build_id: buildId, created_at: buildId.getTimestamp() })));
} catch (error) {
   // error will be caught here and propagated
}

Reply:
Looks fine to me - we're pretty deep in nested awaits here, so makes sense to preserve error context by `return await`. Good change!